New build ends up in new iso upload every time - "[ERROR] Remote command exited with '127': filesum -c"

Everytime a build of an image is started there will be a new upload of the same iso file. The logfile just says:
`[ERROR] Remote command exited with '127': filesum -c`

- Packer version from `packer version`
1.4.1

- Target: VMware ESX 6.x

It was working in former versions of packer.

--- a/common/iso_config.go
+++ b/common/iso_config.go
@@ -58,13 +58,19 @@ func (c *ISOConfig) Prepare(ctx *interpolate.Context) (warnings []string, errs [
 	}
 
 	if c.ISOChecksumURL != "" {
-		if strings.HasSuffix(strings.ToLower(c.ISOChecksumURL), ".iso") {
-			errs = append(errs, fmt.Errorf("Error parsing checksum:"+
-				" .iso is not a valid checksum extension"))
+		if c.ISOChecksum != "" {
+			warnings = append(warnings, "You have provided both an "+
+				"iso_checksum and an iso_checksum_url. Discarding the "+
+				"iso_checksum_url and using the checksum.")
+		} else {
+			if strings.HasSuffix(strings.ToLower(c.ISOChecksumURL), ".iso") {
+				errs = append(errs, fmt.Errorf("Error parsing checksum:"+
+					" .iso is not a valid checksum extension"))
+			}
+			// go-getter auto-parses checksum files
+			c.ISOChecksumType = "file"
+			c.ISOChecksum = c.ISOChecksumURL
 		}
-		// go-getter auto-parses checksum files
-		c.ISOChecksumType = "file"
-		c.ISOChecksum = c.ISOChecksumURL
 	}
 
 	if c.ISOChecksum == "" {
@@ -86,11 +92,12 @@ func (c *ISOConfig) Prepare(ctx *interpolate.Context) (warnings []string, errs [
 			Getters: getter.Getters,
 		}
 		cksum, err := gc.ChecksumFromFile(c.ISOChecksumURL, u)
-		if err != nil {
+		if cksum == nil || err != nil {
 			errs = append(errs, fmt.Errorf("Couldn't extract checksum from checksum file"))
+		} else {
+			c.ISOChecksumType = cksum.Type
+			c.ISOChecksum = hex.EncodeToString(cksum.Value)
 		}
-		c.ISOChecksumType = cksum.Type
-		c.ISOChecksum = hex.EncodeToString(cksum.Value)
 	}
 
 	return warnings, errs

--- a/common/iso_config.go
+++ b/common/iso_config.go
@@ -1,10 +1,15 @@
 package common
 
 import (
+	"encoding/hex"
 	"errors"
 	"fmt"
+	"log"
+	"net/url"
+	"os"
 	"strings"
 
+	getter "github.com/hashicorp/go-getter"
 	"github.com/hashicorp/packer/template/interpolate"
 )
 
@@ -64,6 +69,28 @@ func (c *ISOConfig) Prepare(ctx *interpolate.Context) (warnings []string, errs [
 
 	if c.ISOChecksum == "" {
 		errs = append(errs, fmt.Errorf("A checksum must be specified"))
+	}
+	if c.ISOChecksumType == "file" {
+		u, err := url.Parse(c.ISOUrls[0])
+		wd, err := os.Getwd()
+		if err != nil {
+			log.Printf("get working directory: %v", err)
+			// here we ignore the error in case the
+			// working directory is not needed.
+		}
+		gc := getter.Client{
+			Dst:     "no-op",
+			Src:     u.String(),
+			Pwd:     wd,
+			Dir:     false,
+			Getters: getter.Getters,
+		}
+		cksum, err := gc.ChecksumFromFile(c.ISOChecksumURL, u)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("Couldn't extract checksum from checksum file"))
+		}
+		c.ISOChecksumType = cksum.Type
+		c.ISOChecksum = hex.EncodeToString(cksum.Value)
 	}
 
 	return warnings, errs

--- a/common/iso_config_test.go
+++ b/common/iso_config_test.go
@@ -75,11 +75,16 @@ func TestISOConfigPrepare_ISOChecksum(t *testing.T) {
 func TestISOConfigPrepare_ISOChecksumURLBad(t *testing.T) {
 	i := testISOConfig()
 	i.ISOChecksumURL = "file:///not_read"
+	i.ISOChecksum = "shouldoverride"
 
 	// Test ISOChecksum overrides url
 	warns, err := i.Prepare(nil)
-	if len(warns) > 0 && len(err) > 0 {
-		t.Fatalf("bad: %#v, %#v", warns, err)
+	if len(warns) != 1 {
+		t.Fatalf("Bad: should have warned because both checksum and " +
+			"checksumURL are set.")
+	}
+	if len(err) > 0 {
+		t.Fatalf("Bad; should have warned but not errored.")
 	}
 
 	// Test that we won't try to read an iso into memory because of a user

--- a/go.mod
+++ b/go.mod
@@ -44,7 +44,7 @@ require (
 	github.com/hashicorp/errwrap v1.0.0
 	github.com/hashicorp/go-checkpoint v0.0.0-20171009173528-1545e56e46de
 	github.com/hashicorp/go-cleanhttp v0.5.0
-	github.com/hashicorp/go-getter v1.3.1-0.20190627194820-3fc511e2f341
+	github.com/hashicorp/go-getter v1.3.1-0.20190627223108-da0323b9545e
 	github.com/hashicorp/go-multierror v1.0.0
 	github.com/hashicorp/go-oracle-terraform v0.0.0-20181016190316-007121241b79
 	github.com/hashicorp/go-retryablehttp v0.5.2 // indirect

--- a/go.mod
+++ b/go.mod
@@ -44,7 +44,7 @@ require (
 	github.com/hashicorp/errwrap v1.0.0
 	github.com/hashicorp/go-checkpoint v0.0.0-20171009173528-1545e56e46de
 	github.com/hashicorp/go-cleanhttp v0.5.0
-	github.com/hashicorp/go-getter v1.2.0
+	github.com/hashicorp/go-getter v1.3.1-0.20190627194820-3fc511e2f341
 	github.com/hashicorp/go-multierror v1.0.0
 	github.com/hashicorp/go-oracle-terraform v0.0.0-20181016190316-007121241b79
 	github.com/hashicorp/go-retryablehttp v0.5.2 // indirect

